platform-core-module/src/main/java/com/softicar/platform/core/module/program/QueuedProgramExecutionDaemon.java

```diff
@@ -102,6 +116,29 @@ private void terminate(Collection<IRunnableThread<ProgramExecutionRunnable>> run
 		}
 	}
 
+	private void checkMaximumRuntime(AGProgram program, AGProgramExecution currentExecution) {
+
+		var scheduledExecution = AGScheduledProgramExecution.getByAGUuid(program.getProgramUuid());
+		if (scheduledExecution != null) {
+			var maximumRuntimeInSeconds = scheduledExecution.getMaximumRuntimeInSeconds();
+			if (maximumRuntimeInSeconds.isPresent()) {
+				boolean maximumRuntimeExceeded = DayTime.now().isAfter(currentExecution.getStartedAt().plusSeconds(maximumRuntimeInSeconds.get()));
+				if (maximumRuntimeExceeded && !currentExecution.isExceededMaximumRuntime()) {
+					new SystemEventBuilder(AGSystemEventSeverityEnum.WARNING, CoreI18n.PROGRAM_EXECUTION_EXCEEDED_MAXIMUM_RUNTIME.toString())//
+						.addProperty("program", program.toDisplay().toString())
+						.addProperty("start", currentExecution.getStartedAt().toGermanString())
+						.addProperty("maximumRuntimeSeconds", "" + maximumRuntimeInSeconds)
+						.save();
+					if (scheduledExecution.isAutoKill()) {
+						terminateProgram(program);
+					}
+					currentExecution.reload();
```

Comment:
Can we somehow assume that we're in a `DbTransaction` here? Because `reload` and `save` should be executed in the same transaction.

Reply:
i dont know